.github/workflows/main.yaml

```diff
@@ -0,0 +1,91 @@
+name: CI/CD
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains two jobs named "build" and "container"
+  build:
+    name: Compile and test
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.node-version'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Generate code
+        run: npm run generate
+
+      - name: Compile
+        run: npm run compile
+
+      - name: Lint code
+        run: npm run lint
+
+      - name: Execute tests
+        run: npm test
+
+  container:
+    name: Build container image
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Login to GitHub Package
+        uses: docker/login-action@v2
+        with:
+          registry: 'ghcr.io'
```

Comment:
What do you think to set a variable with this value and use the variable here? It can be useful too on line `60` of this file.